packages/hyperion-autologging/src/ALUIEventPublisher.ts

```diff
@@ -0,0 +1,191 @@
+/**
+ * (c) Meta Platforms, Inc. and affiliates. Confidential and proprietary.
+ *
+ * @flow strict-local
+ * @format
+ * @oncall am_logging
+ */
+
+'use strict';
+import { Hook } from "@hyperion/hook";
+import { Flowlet } from "@hyperion/hyperion-flowlet/src/Flowlet";
+import { FlowletManager } from "@hyperion/hyperion-flowlet/src/FlowletManager";
+import {TimedTrigger} from "@hyperion/hyperion-util/src/TimedTrigger";
+import { getElementName, getInteractable, trackInteractable, trackSynthetic } from "./ALInteractableDOMElement";
+import { DataType, FlowletManagerType, FlowletType } from "./ALSurface";
+import { getSurfacePath } from "./ALSurfaceUtils";
+import {getOrSetAutoLoggingID} from "@hyperion/hyperion-util/src/ALIDUtils";
+import { Channel } from "@hyperion/hook/src/Channel";
+
+export type ALUIEventCaptureData = Readonly<{
+  event: string,
+  element: Element,
+  isTrusted: boolean,
+  flowlet: Flowlet,
+  captureTimestamp: number,
+  surface: string | null,
+  elementName: string | null,
+}>;
+
+export type ALUIEventBubbleData = Readonly<{
+  event: string,
+  element: Element,
+  isTrusted: boolean,
+  bubbleTimestamp: number,
+}>;
+
+export type ALUIEventData = Readonly<{
+  event: string,
+  element: Element,
+  isTrusted: boolean,
+  flowlet: Flowlet,
+  eventIndex: number,
+  eventTimestamp: number,
+  autoLoggingID: string,
+  surface?: string,
+}>;
+
+export type ALChannelUIEvent = Readonly<{
+  al_ui_event_capture: [ALUIEventCaptureData],
+  al_ui_event_bubble: [ALUIEventBubbleData],
+  al_ui_event: [ALUIEventData],
+}>;
+
+export const onALUIEventCapture: Hook<(ALUIEventCaptureData) => void> =
+  new Hook();
+export const onALUIEventBubble: Hook<(ALUIEventBubbleData) => void> =
+  new Hook();
+export const onALUIEvent: Hook<(ALUIEventData) => void> =
+  new Hook();
```

Comment:
We can actually remove these entirely, and just expect a channel to sent the message. They are sort of redundant.